circuits/cpp/src/aztec3/circuits/abis/private_circuit_public_inputs.hpp

```diff
@@ -242,6 +258,8 @@ void write(std::vector<uint8_t>& buf, PrivateCircuitPublicInputs<NCT> const& pri
     write(buf, pis.historic_l1_to_l2_messages_tree_root);
 
     write(buf, pis.contract_deployment_data);
+    write(buf, pis.chain_id);
```

Comment:
While we are here we could probably alter this to use the MSGPACK_FIELDS() macro

Reply:
Seems like it needs changes quite a few places to use `MSGPACK_FIELDS()` consistently (in other part os the code as well), so might be better to put that off to a separate cleanup PR.